### PR TITLE
Update server-tls.mdx

### PR DESCRIPTION
**Pull request by vanphan24**
Added k8s auth role for client
Added to Consul yaml file: tls.enableAutoEncrypt: true
Fixed name of CA policy: policies=ca-policy

### CODE REVIEW

File: website/content/docs/k8s/installation/vault/server-tls.mdx [hunk 1]
```diff
@@ -102,8 +102,9 @@ echo allowed_domains=\"$DATACENTER.consul, $NAME-server, $NAME-server.$NAMESPACE
 
 Prior to creating Vault auth roles for the Consul server and the Consul components, ensure that the Vault Kubernetes auth method is enabled as described in [Vault Kubernetes Auth Method](/docs/k8s/installation/vault#vault-kubernetes-auth-method).
 
-Finally, two Kubernetes auth roles need to be created, one for the Consul servers and one for the Consul components:
+Finally, three Kubernetes auth roles need to be created, one for the Consul servers, one for the Consul clients, and one for the Consul components.
 
+Role for Consul servers:
 ```shell-session
 vault write auth/kubernetes/role/consul-server \
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/installation/vault/server-tls.mdx:109]:
```suggestion
$ vault write auth/kubernetes/role/consul-server \
```
Commands should be prefixed with $

File: website/content/docs/k8s/installation/vault/server-tls.mdx [hunk 2]
```diff
@@ -121,11 +122,30 @@ you can run:
 -> **Note:** Should you enable other supported features such as gossip-encryption be sure to append additional policies to
 the Kube auth role in a comma separated value e.g.  `policies=consul-server,consul-gossip`
 
+Role for Consul clients:
+```shell-session
+vault write auth/kubernetes/role/consul-client \
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/installation/vault/server-tls.mdx:127]:
```suggestion
$ vault write auth/kubernetes/role/consul-client \
```

File: website/content/docs/k8s/installation/vault/server-tls.mdx [hunk 3]
```diff
@@ -121,11 +122,30 @@ you can run:
...
+    bound_service_account_names=<Consul client service account>  \
+    bound_service_account_namespaces=default \
+    policies=ca-policy \
+    ttl=1h
+```
+
+To find out the service account name of the Consul client
+you can run:
+```shell-session
+ helm template --release-name <your release name> -s templates/client-serviceaccount.yaml hashicorp/consul
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/installation/vault/server-tls.mdx:137]:
```suggestion
 $ helm template --release-name <your release name> -s templates/client-serviceaccount.yaml hashicorp/consul
```

File: website/content/docs/k8s/installation/vault/server-tls.mdx [hunk 4]
```diff
@@ -121,11 +122,30 @@ you can run:
...
+```
+
+-> **Note:** Should you enable other supported features such as gossip-encryption be sure to append additional policies to
+the Kube auth role in a comma separated value e.g.  `policies=ca-policy,consul-gossip`
+
+Role for CA components:
 ```shell-session
 vault write auth/kubernetes/role/consul-ca \
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/installation/vault/server-tls.mdx:145]:
```suggestion
$ vault write auth/kubernetes/role/consul-ca \
```

File: website/content/docs/k8s/installation/vault/server-tls.mdx [hunk 5]
```diff
@@ -121,11 +122,30 @@ you can run:
...
+    ttl=1h
+```
+
+To find out the service account name of the Consul client
+you can run:
+```shell-session
+ helm template --release-name <your release name> -s templates/client-serviceaccount.yaml hashicorp/consul
+```
+
+-> **Note:** Should you enable other supported features such as gossip-encryption be sure to append additional policies to
```
Comment (karl-cardenas-coding) [inline website/content/docs/k8s/installation/vault/server-tls.mdx:140]:
```suggestion
-> **Note:** Should you enable other supported features such as gossip-encryption, ensure you append additional policies to
```
Removed passive language

**Review by karl-cardenas-coding — APPROVED** [review]
Looks good Van, just some minor fixes and suggestions.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/550950.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 4b730392d6d002c2cb684f81dc8ba9db4873f202 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 4b730392d6d002c2cb684f81dc8ba9db4873f202 onto `release/1.11.x` succeeded!

**Pull request merged**